cpp/include/cudf/copying.hpp

```diff
@@ -143,27 +143,20 @@ std::unique_ptr<column> reverse(
  * A negative value `i` in the `scatter_map` is interpreted as `i+n`, where `n`
  * is the number of rows in the `target` table.
  *
- * @throws cudf::logic_error if `check_bounds == true` and an index exists in
- * `scatter_map` outside the range `[-n, n)`, where `n` is the number of rows in
- * the target table. If `check_bounds == false`, the behavior is undefined.
```

Comment:
I suggest we keep some of this to document that the behavior is undefined for indices outside of `[-n,-n)`